Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Cursors/BaseCursor.cs

```diff
@@ -141,7 +141,13 @@ public virtual void OnSourceDetected(SourceStateEventData eventData)
                     if (eventData.InputSource.Pointers[i].PointerId == Pointer.PointerId)
                     {
                         visibleSourcesCount++;
-                        SetVisibility(true);
+
+                        if (visibleSourcesCount == 1 &&
```

Comment:
curious... is this needed to fix this? or is it a piece of the gesture pr that is related?

Reply:
It's part of the fix.